alvr/adb/src/lib.rs

```diff
@@ -0,0 +1,142 @@
+// https://android.googlesource.com/platform/packages/modules/adb/+/refs/heads/main/docs/user/adb.1.md
+
+pub mod connection_state;
+pub mod device;
+pub mod forwarded_port;
+pub mod transport_type;
+
+use std::{io::Cursor, path::PathBuf, process::Command, time::Duration};
+
+use const_format::formatcp;
+use device::Device;
+use forwarded_port::ForwardedPort;
+use zip::ZipArchive;
+
+#[cfg(not(windows))]
+const ADB_EXECUTABLE: &str = "adb";
+#[cfg(windows)]
+const ADB_EXECUTABLE: &str = "adb.exe";
+
+// https://developer.android.com/tools/releases/platform-tools#revisions
+// NOTE: At the time of writing this comment, the revisions section above
+// shows the latest version as 35.0.2, but the latest that can be downloaded
+// by specifying a version is 35.0.0
+const PLATFORM_TOOLS_VERSION: &str = "-latest"; // E.g. "_r35.0.0"
+
+#[cfg(target_os = "linux")]
+const PLATFORM_TOOLS_OS: &str = "linux";
+#[cfg(target_os = "macos")]
+const PLATFORM_TOOLS_OS: &str = "darwin";
+#[cfg(windows)]
+const PLATFORM_TOOLS_OS: &str = "windows";
+
+const PLATFORM_TOOLS_URL: &str = formatcp!("https://dl.google.com/android/repository/platform-tools{PLATFORM_TOOLS_VERSION}-{PLATFORM_TOOLS_OS}.zip");
+
+const REQUEST_TIMEOUT: Duration = Duration::from_secs(30);
```

Comment:
I would never use any wait longer than 5 seconds. if this contains also the time to download the binary then you should rework the code a bit.

Reply:
I think you should use a timeout for checking if download started, and then monitor the download, then use timeout for when the download gets stuck. we cannot assume download will take less than 30 seconds. Also you should add a dashboard event for adb progress